core/dbt/parser/analysis.py

```diff
@@ -1,14 +1,27 @@
+import os
+from typing import Dict, Any
 
+from dbt.contracts.graph.parsed import ParsedAnalysisNode, ParsedRPCNode
 from dbt.parser.base_sql import BaseSqlParser
-import os
 
 
 class AnalysisParser(BaseSqlParser):
     @classmethod
     def get_compiled_path(cls, name, relative_path):
         return os.path.join('analysis', relative_path)
 
+    def parse_from_dict(
+        self,
+        parsed_dict: Dict[str, Any]
+    ) -> ParsedAnalysisNode:
+        """Given a dictionary, return the parsed entity for this parser"""
+        return ParsedAnalysisNode.from_dict(parsed_dict)
 
-class RPCCallParser(AnalysisParser):
+
+class RPCCallParser(BaseSqlParser):
     def get_compiled_path(cls, name, relative_path):
         return os.path.join('rpc', relative_path)
+
+    def parse_from_dict(self, parsed_dict: Dict[str, Any]) -> ParsedRPCNode:
+        """Given a dictionary, return the parsed entity for this parser"""
+        return ParsedRPCNode.from_dict(parsed_dict)
```

Comment:
all of these duplicated `parse_from_dict`s actually feel like an unfortunate side effect of using type annotations. is there a way to use a "generic" here and specify which node type each Parser generates?

this case actually doesn't feel like the biggest deal. this code is simple. but i can see this being troublesome in the future when we want to inherit functionality and have to redefine the entire method to annotate the return type

Reply:
Yeah. The real thing I want is for `MacrosKnownParser` (or something?) to subclass `Generic[T]` where `T` is something ParsedNode-like and then subclasses would be like:

```
class AnalysisParser(MacrosKnownParser[ParsedAnalysisNode]):
   ...
```

I just didn't really want to spend a lot of time messing with parsing class definitions in this PR only to rip them out or be deeply inconvenienced in the PRs for #1487 or #1600 where I imagine I'll have to make a ton of changes. `typing.Generic` can result in some seriously weird errors because it does some metaclass craziness.